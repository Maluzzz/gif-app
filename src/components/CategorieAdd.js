import React, {useState} from "react"
import PropTypes from 'prop-types'

export const CategorieAdd = ({setCategories}) => {
  const [text, setText] = useState('')
  const handleChange = (e) => {
    setText(e.target.value)
    
  }
  const handleSubmit = (e) => {
      e.preventDefault()
      if(text.trim().length !== 0 ){
      setCategories(cat => [text, ...cat])
      setText('')
    }
  }
  return (
   
      <form onSubmit={handleSubmit} >
        <input type="text" value={text} onChange={handleChange} placeholder="Search for Gifs"/>
        <button >Search</button>
      </form>
  
  )
}

CategorieAdd.protoTypes = {
    setCategories: PropTypes.func.isRequired
}