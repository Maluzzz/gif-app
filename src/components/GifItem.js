import React from 'react'
import PropTypes from 'prop-types'

export default function GifItem({ title, url }) {
    return (
        <>
            <div className="card-image animate__animated  animate__pulse">
                <img src={url} alt={title}></img>
                <p className='gif-title'>{title}</p>
            </div>
        </>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
