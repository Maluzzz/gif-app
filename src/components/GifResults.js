import React  from 'react'
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs'
import GifItem from './GifItem'



export const GifResults = ({category}) => {

  const {data: imgs,loading} = useFetchGifs(category)


  return (
    <>
      <h3>{category}</h3>
      {loading && <p>'Cargando'</p>}
      <div className='card-container'>
        {imgs.map((img) => (
          <GifItem key={img.id} {...img} />
        ))}
      </div>
    </>
  )
}

GifResults.propTypes= {
  category: PropTypes.string.isRequired
}
