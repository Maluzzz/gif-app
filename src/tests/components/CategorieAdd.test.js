import '@testing-library/jest-dom'

import { shallow } from 'enzyme'
import { CategorieAdd } from '../../components/CategorieAdd'


describe('testing Component CategorieAdd', () => {

      const setCategories = jest.fn()
      let wrapper = shallow(<CategorieAdd setCategories={setCategories} />)

      beforeEach(() => {
            jest.clearAllMocks()
            wrapper = shallow(<CategorieAdd setCategories={setCategories} />)
      })

      test('should render the component', () => {

            expect(wrapper).toMatchSnapshot()
      })

      test('should change the input', () => {

            const input = wrapper.find('input')
            input.simulate('change', { target: { value: 'Hellow word' } })
            //TO.DO CHECK INPUT VALUE VALUE IS HELLOW WORLD

      })

      test('shouldnt send the submit with setCategories', () => {

            const form = wrapper.find('form')
            form.simulate('submit', { preventDefault() { } })
            expect(setCategories).not.toHaveBeenCalled()

      })

      test('should send the submit, call the setCategories and clean the inputText', () => {


            const input = wrapper.find('input')
            input.simulate('change', { target: { value: 'Hellow word' } })
            const form = wrapper.find('form')
            form.simulate('submit', { preventDefault() { }, text: 'Hellow word' })

            expect(setCategories).toHaveBeenCalled()
            expect(setCategories).toHaveBeenCalledWith( expect.any(Function) )
            expect(wrapper.find('input').props().value).toBe('')

      })

})
