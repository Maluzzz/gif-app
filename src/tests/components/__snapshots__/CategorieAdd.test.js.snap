// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Component CategorieAdd should render the component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search for Gifs"
    type="text"
    value=""
  />
  <button>
    Search
  </button>
</form>
`;
